### PR TITLE
feat(server): crud products

### PR DESCRIPTION
**Pull request by akinncar**
Create Product to a Store
```js
mutation ProductCreateMutation($input: ProductCreateInput!) {
	ProductCreate(input: $input) {
		product {
			name
			description
			pictureUrl
			points
			storeId
		}
		error
	}
}
```
```js
{
	"input": {
		"name": "X-Bacon",
		"description": "Feito de Bacon",
		"pictureUrl": "https://embutidosbonatti.ind.br/temp/BIN_57_V9Fb0BwK.jpg",
		"points": 5,
		"storeId": "61aeb2262dcb61ccd9f53ec2"
	}
}
```

List Products by Store
```js
query ProductListByStoreQuery($storeId: ID!) {
  products: productsByStoreId(storeId: $storeId) {
    edges {
      node {
        _id
        name
        description
        pictureUrl
        points
        storeId
      }
    }
  }
}
```

### CODE REVIEW

**Comment by akinncar** [comment]
how to link `storeId` in `product` entity to a document from `store` entity?

File: packages/server/src/modules/product/ProductModel.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      required: false,
+    },
+    points: {
+      type: Number,
+      required: false,
+    },
+    storeId: {
+      type: ObjectId,
+      required: false,
+    },
```
Comment (NicolasLopes7) [inline packages/server/src/modules/product/ProductModel.ts:27]: you need to and a ref field to relate these collections.

```suggestion
    storeId: {
      type: ObjectId,
      ref: 'Store',
      required: false,
    },

File: packages/server/src/modules/product/ProductModel.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import mongoose, { Document, Model } from 'mongoose';
+
+const { ObjectId } = mongoose.Schema.Types;
+const Schema = mongoose.Schema;
+
+const ProductSchema = new Schema(
```
Comment (daniloab) [inline packages/server/src/modules/product/ProductModel.ts:6]: declaring the interface before you can use with schema like `new Schema< IProduct>(`

File: packages/server/src/modules/product/ProductType.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { GraphQLID, GraphQLObjectType, GraphQLString, GraphQLInt } from 'graphql';
+import { globalIdField } from 'graphql-relay';
+import { objectIdResolver } from '@entria/graphql-mongo-helpers';
+
+// import TeamType from "../team/TeamType";
```
Comment (daniloab) [inline packages/server/src/modules/product/ProductType.ts:5]: remove comments

File: packages/server/src/modules/product/mutations/ProductCreate.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    points: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+    storeId: {
+      type: new GraphQLNonNull(GraphQLID),
+    },
+  },
+  mutateAndGetPayload: async ({ id, name, description, pictureUrl, points, storeId }) => {
+    const product = await new Product({
```
Comment (daniloab) [inline packages/server/src/modules/product/mutations/ProductCreate.ts:30]: probably you need to do some checks like

```js
const store = ....({ _id: storeId })

if(!store) { return { error: 'store not found' }}
```

File: packages/server/src/schema/QueryType.ts [hunk 5]
```diff
@@ -52,5 +57,15 @@ export default new GraphQLObjectType({
       resolve: async (_, args, context) =>
         StoreLoader.load(context, args.id),
     },
+    productsByStoreId: {
```
Comment (daniloab) [inline packages/server/src/schema/QueryType.ts:60]: avoid rest names

it is better get this from a node interface from Store then get a list of products inside of this store

Comment (daniloab) [reply packages/server/src/schema/QueryType.ts:60]:
```
StoreType: {
  ...,
  products: {} // filter the products by storeId
}
```

then, on frontend you do a request by node interface

**Pull request merged**